addons/hw_drivers/controllers/driver.py

```diff
@@ -1,432 +1,354 @@
+        with open(path, 'r') as f:
+            return f.readline().strip('\n')
+    return ''
 
-    def get_connection(self):
-        pass
+def get_odoo_server_url():
+    return read_file_first_line('odoo-remote-server.conf')
 
-    def action(self, action):
-        pass
+def get_token():
+    return read_file_first_line('token')
 
 #----------------------------------------------------------
-# Usb drivers
+# Controllers
 #----------------------------------------------------------
-usbdrivers = []
-drivers = {}
 
-class UsbMetaClass(type):
-    def __new__(cls, clsname, bases, attrs):
-        newclass = super(UsbMetaClass, cls).__new__(cls, clsname, bases, attrs)
-        usbdrivers.append(newclass)
-        return newclass
-
-class USBDriver(Driver,metaclass=UsbMetaClass):
-    def __init__(self, dev):
-        super(USBDriver, self).__init__()
-        self.dev = dev
-        self.value = ""
-
-    def get_name(self):
-        lsusb = str(subprocess.check_output('lsusb')).split("\\n")
-        for usbpath in lsusb:  # Should filter on usb devices or inverse loops?
-            device = self.dev
-            if "%04x:%04x" % (device.idVendor, device.idProduct) in usbpath:
-                return usbpath.split("%04x:%04x" % (device.idVendor, device.idProduct))[1]
-        return str(device.idVendor) + ":" + str(device.idProduct)
-
-    def get_connection(self):
-        return 'direct'
-
-    def value(self):
-        return self.value
+class StatusController(http.Controller):
+    @http.route('/hw_drivers/action', type='json', auth='none', cors='*', csrf=False)
+    def action(self, session_id, device_id, data):
+        iot_device = iot_devices.get(device_id)
+        if iot_device:
+            iot_device.action(session_id, data)
+            return True
+        return False
+
+    @http.route('/hw_drivers/event', type='json', auth='none', cors='*', csrf=False)
+    def event(self, listener):
+        """
+        listener is a dict in witch there are a sessions_id and a dict of device_id to listen
+        """
+        req = DeviceManager.add_request(listener)
+        if req['event'].wait(58):
+            req['event'].clear()
+            req['queue']['session_id'] = req['session_id']
+            return req['queue']
 
+    @http.route('/hw_drivers/box/connect', type='json', auth='none', cors='*', csrf=False)
+    def connect_box(self, token):
+        """
+        This route is called when we want that a IoT Box will be connected to a Odoo DB
+        token is a base 64 encoded string and have 2 argument separate by |
+        1 - url of odoo DB
+        2 - token. This token will be compared to the token of Odoo. He have 1 hour lifetime
+        """
+        server = get_odoo_server_url()
+        if server:
+            return _('This IoTBox has already been connected')
+        url, token = token.split('|')
+        path = os.getenv("HOME") + '/odoo/addons/point_of_sale/tools/posbox/configuration/connect_to_server.sh'
```

Comment:
if the home env variable doesn't exist, you'll get a `TypeError` here
you'd probably better return early if home isn't set